### PR TITLE
Adjust customdummy plugin in Docs to new API

### PR DESCRIPTION
**Pull request by klecki**
Signed-off-by: Krzysztof Lecki <klecki@nvidia.com>

#### Why we need this PR?
- It fixes a bug in Customdummyplugin example in docs

#### What happened in this PR?
 - Explain solution of the problem, new feature added.
The Customdummyplugin was not adjusted to new SetupAPI
 - What was changed, added, removed?
Adjusting it to new Setup API.
Extended description about what the operator does.
 - What is most important part that reviewers should focus on?
Everything
 - Was this PR tested? How?
L1 jupyter test will be run
 - Were docs and examples updated, if necessary?
Yes

**JIRA TASK**: [DALI-1009]

### CODE REVIEW

**Comment by klecki** [comment]
!build

File: docs/examples/extend/create_a_custom_operator.ipynb [hunk 1]
```diff
@@ -341,10 +352,10 @@
       "-- Looking for pthread_create in pthread\n",
       "-- Looking for pthread_create in pthread - found\n",
       "-- Found Threads: TRUE  \n",
-      "-- Found CUDA: /usr/local/cuda (found suitable version \"9.0\", minimum required is \"8.0\") \n",
+      "-- Found CUDA: /usr/local/cuda (found suitable version \"10.1\", minimum required is \"8.0\") \n",
       "-- Configuring done\n",
       "-- Generating done\n",
-      "-- Build files have been written to: /home/dali/git/dali/docs/examples/extend/customdummy/build\n",
+      "-- Build files have been written to: /home/klecki/work/salvador/dali/docs/examples/extend/customdummy/build\n",
```
Comment (jantonguirao) [inline docs/examples/extend/create_a_custom_operator.ipynb]: revert this line

File: docs/examples/extend/create_a_custom_operator.ipynb [hunk 2]
```diff
@@ -27,7 +27,7 @@
    "source": [
     "## Operator definition\n",
     "\n",
-    "First we define the operator in a header file "
+    "First we define the operator in a header file. We will create an Operator that copies it's only input to output. We can provide a common implementation for both backends of `CanInferOutputs` and `SetupImpl`. `SetupImpl` provides shape and type description of the output based on the input and `CanInferOutputs` informs executor that Operator will provide that output description."
```
Comment (mzient) [inline docs/examples/extend/create_a_custom_operator.ipynb]:
```suggestion
    "First we define the operator in a header file. We will create an Operator that copies its only input to the output. The implementation of `CanInferOutputs` and `SetupImpl` can be shared across backends. `SetupImpl` provides shape and type description of the output based on the input and `CanInferOutputs` informs the executor that the Operator can provide that output description for the whole batch before executing `RunImpl`."
```

Comment (klecki) [reply docs/examples/extend/create_a_custom_operator.ipynb]: Done

File: docs/examples/extend/create_a_custom_operator.ipynb (hunk 1 above)
Comment (klecki) [reply docs/examples/extend/create_a_custom_operator.ipynb]: Done

**Comment by klecki** [comment]
!build

**Pull request merged**